### PR TITLE
 :warning: Update kubebuilder cli and plugin v3+ to use go version 1.15

### PR DESCRIPTION
**Pull request by rashmigottipati**
**Description**

- Update go version from 1.13 to 1.15 

**Note**
Here's the link to update the infra to run the CI jobs as well: kubernetes/test-infra#19527

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @rashmigottipati, 

You need to follow the instructions in https://github.com/kubernetes-sigs/kubebuilder/pull/1716#issuecomment-705920674
/ok-to-test

**Comment by camilamacedo86** [comment]
Hi @DirectXMan12, 

Controller-runtime has updated o use 1.15 already. Have you any objection to this change? Just to confirm. 
For me it is OK. 

/approved

**Comment by rashmigottipati** [comment]
I signed it

**Comment by rashmigottipati** [comment]
@camilamacedo86 thanks for your input Camila. I've created this PR in kubernetes/test-infra updating the go version in https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kubebuilder/kubebuilder-presubmits.yaml#L13 to 1.15. 
Here's the link: https://github.com/kubernetes/test-infra/pull/19527

File: pkg/plugin/internal/util/go_version.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func checkGoVersion(verStr string) error {
 	}
 
 	if major < 1 || minor < 13 {
```
Comment (estroz) [inline pkg/plugin/internal/util/go_version.go:71]:
```suggestion
	if major < 1 || minor < 15 {
```

File: pkg/plugin/v2/scaffolds/internal/templates/dockerfile.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (f *Dockerfile) SetTemplateDefaults() error {
 }
 
 const dockerfileTemplate = `# Build the manager binary
-FROM golang:1.13 as builder
```
Comment (estroz) [inline pkg/plugin/v2/scaffolds/internal/templates/dockerfile.go:42]: We shouldn't be changing v2's Go version unless we account for plugin version in `checkGoVersion()`.

**Review by estroz — CHANGES_REQUESTED** [review]
Bumping the v2 plugin to go 1.15 is a breaking change because the binary will fail for previously scaffolded projects (in `checkGoVersion()`). I recommend only bumping in v3.

File: pkg/plugin/internal/util/go_version.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func checkGoVersion(verStr string) error {
 	}
```
Comment (estroz) [inline pkg/plugin/internal/util/go_version.go:69]: Go version checks should be refactored into versioned plugin packages such that the binary can accept multiple go versions.

Comment (camilamacedo86) [reply pkg/plugin/internal/util/go_version.go:69]: good catcher.

**Comment by camilamacedo86** [comment]
Hi @estroz, 

> Bumping the v2 plugin to go 1.15 is a breaking change because the binary will fail for previously scaffolded projects (in checkGoVersion()). I recommend only bumping in v3.

I agree that is a breaking change. However, in this case, I think it is fine and we need to move forward with both otherwise it will be hard to keep maintained and used. See that: 

- It is very strange to ask for the user use 1.13 and if would like to scaffold --plugins=v3 then uninstall/install another go version. 
- Also, keep 2 diffs go versions will make hard to us collab work with
- the tests are NOT implemented to accommodate 2 go versions for example, how we can use teh go 1.13 to gen the project-v2 and go 1.15 for the project-v3?

So, IHMO we cannot have go 1.13 and go 1.15. 
c/c @rashmigottipati

File: pkg/plugin/v2/scaffolds/internal/templates/dockerfile.go (hunk 2 above)
Comment (rashmigottipati) [reply pkg/plugin/v2/scaffolds/internal/templates/dockerfile.go]: Reverted v2's Go version to 1.13.

**Comment by estroz** [comment]
@rashmigottipati can you add a comment above `checkGoVersion()` as to why it wasn't updated? (because the kubebuilder binary only cares that the go binary supports modules, which were on by default in go 1.13)

**Comment by rashmigottipati** [comment]
thanks @estroz, just added a comment as per our discussion.

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**